doc/semantics.md

```diff
@@ -48,44 +60,80 @@ type they were created with. The following are examples:
 - 1 matches `Byte`, `Short`, `Int`, `Float`, `Double`
 - 128 (`> Byte.MaxValue`) matches `Short`, `Int`, `Float`, `Double`
 - 32768 (`> Short.MaxValue`) matches `Int`, `Float`, `Double`
+- 2147483647 matches `Int`, `Double` if strict-floats are enabled,
+  otherwise `Float` as well
```

Comment:
If strict-floats are disabled, any float could assume a value which is in fact not representable in 32 bits, but in 64 bits. So any `number` must match `Float`.